testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -331,6 +346,41 @@ public void testReferencedTablesWithViews()
         assertThat(column.getMasks()).isEmpty();
     }
 
+    @Test
+    public void testReferencedTablesWithMaterializedViews()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("SELECT test_column FROM mock.default.test_materialized_view", 2);
+
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+
+        List<TableInfo> tables = event.getMetadata().getTables();
+        assertThat(tables).hasSize(2);
```

Comment:
why 2?

Reply:
Two tables are involved .. One is the materialized view table and other is the source table of materialized view

Reply:
Technically, one doesn't access MV table. It's just being analyzed.

Reply:
`event.getMetadata().getTables()` gives the table details which are being analyzed...(so it gives us the MV tables and tables which is used by MV table)